shared/editor/nodes/Mention.ts

```diff
@@ -0,0 +1,187 @@
+import Token from "markdown-it/lib/token";
+import { InputRule } from "prosemirror-inputrules";
+import { NodeSpec, Node as ProsemirrorNode, NodeType } from "prosemirror-model";
+import { EditorState, TextSelection, Plugin } from "prosemirror-state";
+import { MarkdownSerializerState } from "../lib/markdown/serializer";
+import { run } from "../plugins/BlockMenuTrigger";
+import isInCode from "../queries/isInCode";
+import { Dispatch, EventType } from "../types";
+import Node from "./Node";
+
+const OPEN_REGEX = /(?:^|\s)@([a-zA-Z+-]+)?$/;
+const CLOSE_REGEX = /(?:^|\s)@(([0-9a-zA-Z_+-]*\s+)|(\s+[0-9a-zA-Z_+-]+)|[^0-9a-zA-Z_+-]+)$/;
+
+export default class Mention extends Node {
+  get name() {
+    return "mention";
+  }
+
+  get schema(): NodeSpec {
+    return {
+      attrs: {
+        style: {
+          default: "",
+        },
+        "data-type": {
+          default: undefined,
+        },
+        "data-id": {
+          default: undefined,
+        },
+        label: {
+          default: undefined,
+        },
+      },
+      inline: true,
+      content: "text*",
+      marks: "",
+      group: "inline",
+      atom: true,
+      parseDOM: [
+        {
+          tag: `span.${this.name}`,
+          preserveWhitespace: "full",
+          getAttrs: (dom: HTMLElement) => ({
+            "data-type": dom.dataset.type,
+            "data-id": dom.dataset.id,
+            label: dom.innerText,
+          }),
+        },
+      ],
+      toDOM: (node) => {
+        return [
+          "span",
+          {
+            class: `${this.name}`,
```

Comment:
@tommoor what style do we expect for this `node`? Currently it's indistinguishable from rest of the document text.

Reply:
Lets make it something like this for now – will definitely revisit before it goes live though:

<img width="117" alt="image" src="https://user-images.githubusercontent.com/380914/218107188-4199a07a-fd37-4a69-8ba7-614b66005f1a.png">